core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/DefaultHealthCheckerContext.java

```diff
@@ -261,11 +283,15 @@ private void rejectIfDestroyed(Object command) {
         }
     }
 
+    @SuppressWarnings("GuardedBy")
     private <T extends Future<U>, U> T add(T future) {
         scheduledFutures.put(future, Boolean.TRUE);
```

Comment:
I'm not really sure if this could be ignored 🤔

Reply:
Yes, it is because this `add` method is called within the lock. Let's add a comment. 😉 
```
// The caller who calls this method must have the lock.
```